src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -49,7 +49,7 @@ struct AggregateFunctionProperties;
   *  (which can be created in some memory pool),
   *  and IAggregateFunction is the external interface for manipulating them.
   */
-class IAggregateFunction
+class IAggregateFunction : public std::enable_shared_from_this<IAggregateFunction>
```

Comment:
If we need to use shared_from_this it almost always means that we do something wrong.

Reply:
Yes, better to replace with `intrusive_ptr` and a static `create` method, because no one needs to create an instance of IAggregateFunction without ptr. But the amount of changes is higher - so better to postpone this change to a next PR.

Also I considered passing the shared_ptr directly into the `getStateType` method. But it's more fragile.

Reply:
@alexey-milovidov let me do that refactoring after JIT pull request will be finished, currently there will be a bunch of conflicts.

Reply:
@kitaisreal Now we are ready for this refactoring :)